### PR TITLE
Enterprise/bug/#389 fix insufficient permissions on enterprise

### PR DESCRIPTION
**Pull request by toan-quach**
Fix insufficient permissions on enterprise due to scenario get_sequence is based on _create in SequenceManager
Changes:
- Split `_create` and `_build_sequence` functions to have better authorization and reuse on enterprise

### CODE REVIEW

**Review by trgiangdo — DISMISSED** [review]
Neat

File: taipy/core/sequence/_sequence_manager.py [hunk 1]
```diff
@@ -171,10 +175,52 @@ def _create(
...
+        sequence = cls._build_sequence(sequence_name, _tasks, subscribers, properties, scenario_id, version)
+        sequence_id = sequence.id
+
         for task in _tasks:
             if sequence_id not in task._parent_ids:
                 task._parent_ids.update([sequence_id])
                 task_manager._set(task)
+
+        if not sequence._is_consistent():
+            raise InvalidSequence(sequence.id)
```
Comment (jrobinAV) [inline taipy/core/sequence/_sequence_manager.py]:
```suggestion
            raise InvalidSequence(sequence_id)
```

**Review by jrobinAV — DISMISSED** [review]
Good catch and fix!

**Review by jrobinAV — COMMENTED** [review]
Looking at the test results and I believe some code is not 3.8 compatible.
Can you check?

**Review by jrobinAV — DISMISSED** [review]
What about renaming Reason to Reasons in plural?

**Review by trgiangdo — DISMISSED** [review]

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**